src/Stratis.Features.SystemContracts/Identifier.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Linq;
+using NBitcoin;
+
+namespace Stratis.Features.SystemContracts
+{
+    /// <summary>
```

Comment:
Embedded contract's `version + contract id` is equivalent in purpose to the `transaction hash` associated with on-chain deployment. Based on this insight a `pseudo-hash` (`uint256`) was introduced together with our BIP 9 integration. Consider using that instead.